src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/transformers/SharedMemoryAccessTransformer.kt

```diff
@@ -270,55 +268,4 @@ internal class SharedMemoryAccessTransformer(
         invokeStatic(Injections::afterRead)
         // STACK: value
     }
-
-    /*
-     * For an array access instruction (either load or store),
-     * tries to obtain the type of the read/written array element.
-     *
-     * If the type can be determined from the opcode of the instruction itself
-     * (e.g., IALOAD/IASTORE) returns it immediately.
-     *
-     * Otherwise, queries the analyzer to determine the type of the array in the respective stack slot.
-     * This is used in two cases:
-     * - for `BALOAD` and `BASTORE` instructions, since they are used to access both boolean and byte arrays;
-     * - for `AALOAD` and `AASTORE` instructions, to get the class name of the array elements.
-     */
-    private fun getArrayElementType(opcode: Int): Type = when (opcode) {
-        // Load
-        IALOAD -> INT_TYPE
-        FALOAD -> FLOAT_TYPE
-        CALOAD -> CHAR_TYPE
-        SALOAD -> SHORT_TYPE
-        LALOAD -> LONG_TYPE
-        DALOAD -> DOUBLE_TYPE
-        BALOAD -> getArrayAccessTypeFromStack(2) ?: BYTE_TYPE
-        AALOAD -> getArrayAccessTypeFromStack(2) ?: OBJECT_TYPE
-        // Store
-        IASTORE -> INT_TYPE
-        FASTORE -> FLOAT_TYPE
-        CASTORE -> CHAR_TYPE
-        SASTORE -> SHORT_TYPE
-        LASTORE -> LONG_TYPE
-        DASTORE -> DOUBLE_TYPE
-        BASTORE -> getArrayAccessTypeFromStack(3) ?: BYTE_TYPE
-        AASTORE -> getArrayAccessTypeFromStack(3) ?: OBJECT_TYPE
-        else -> throw IllegalStateException("Unexpected opcode: $opcode")
-    }
-
-   /*
-    * Tries to obtain the type of array elements by inspecting the type of the array itself.
-    * To do this, the method queries the analyzer to get the type of accessed array
-    * which should lie on the stack.
-    * If the analyzer does not know the type, then return null
-    * (according to the ASM docs, this can happen, for example, when the visited instruction is unreachable).
-    */
-    private fun getArrayAccessTypeFromStack(position: Int): Type? {
-        if (analyzer.stack == null) return null
-        val arrayDesc = analyzer.stack[analyzer.stack.size - position]
-        check(arrayDesc is String)
-        val arrayType = getType(arrayDesc)
-        check(arrayType.sort == ARRAY)
-        check(arrayType.dimensions > 0)
-        return getType(arrayDesc.substring(1))
-    }
```

Comment:
Please unify `SharedMemoryAccessTransformer` and `SnapshotTrackerTransformer` code and then rollback these changes.